components/main-chef-wrapper/platform-lib/cli_helpers.go

```diff
@@ -0,0 +1,154 @@
+package platform_lib
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/chef/chef-workstation/components/main-chef-wrapper/dist"
+)
+
+var rubyenvMap map[string]interface{}
+
+func InitializeRubyMap() {
+	rubyenvMap = UnmarshallRubyEnv()
+}
+
+func PackageHome() string {
+	var packageHomeSet = os.Getenv("CHEF_WORKSTATION_HOME")
+	var packageHome string
+	if len(packageHomeSet) != 0 {
+		packageHome = packageHomeSet
+	} else {
+		packageHome = DefaultPackageName()
+	}
+	return packageHome
+}
+
+func DefaultPackageName() string {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		log.Fatal(err)
+	}
+	if runtime.GOOS == "windows" {
+		return filepath.Join(os.Getenv("LOCALAPPDATA"), dist.WsDirWindows)
+	} else {
+		return filepath.Join(home, dist.WorkstationDir)
+	}
+}
+
+func OmnibusGemRoot() string {
+	gemRoot := ""
+	if rubyenvMap == nil {
+		return gemRoot
+	}
+	data, ok := rubyenvMap["omnibus path"].(map[string]interface{})
+	if ok {
+		gemRoot = data["GEM_ROOT"].(string)
+	}
+	return gemRoot
+}
+
+func RubyExecutable() string {
+	rubyExe := ""
+	if rubyenvMap == nil {
+		return rubyExe
+	}
+	data, ok := rubyenvMap["ruby info"].(map[string]interface{})
+	if ok {
+		rubyExe = data["Executable"].(string)
+	}
+	return rubyExe
+}
+
+func RubyVersion() string {
+	rubyVersion := ""
+	if rubyenvMap == nil {
+		return rubyVersion
+	}
+	data, ok := rubyenvMap["ruby info"].(map[string]interface{})
```

Comment:
we are parsing everything dynamic if we already know what data will come we should use struct.it will save us checking and dynamic casting

Reply:
since JSON data is unstructured, for now so i have used dynamic parsing.

Reply:
ok,got it.